Promote etcd to stable as of 1.19.5 (#2903)
rancher/docs: content/k3s/latest/en/installation/ha-embedded/_index.md

---
title: "High Availability with Embedded DB"
weight: 40
---

> K3s has added support for embedded etcd as of release v1.19.5+k3s1. Versions v1.19.1 thru v1.19.4 provide only experimental support for embedded etcd.

Reviewer: Can we add "full" to distinguish from "experimental support" of version 1.19.1  - 4. 
```suggestion
K3s has added full support for embedded etcd as of release v1.19.5+k3s1. Versions v1.19.1 thru v1.19.4 provide only experimental support for embedded etcd.
```

> In K3s 1.0.0, Dqlite was used as the experimental embedded database. In K3s v1.19.1+, embedded etcd is used.

Reviewer: ```suggestion
In K3s 1.0.0, Dqlite was used as the experimental embedded database. In K3s v1.19.1+, it has been dropped and replaced with etcd.
```

> Please note that upgrades from experimental Dqlite to experimental embedded etcd are not supported. If you attempt an upgrade it will not succeed and data will be lost.

Reviewer 2: should we also mention that dqlite will no longer be supported explicitly

Author: Hmm - it never was technically supported since it was in experimental. Do you think something like this may help?

> In K3s 1.0.0, Dqlite was used as the experimental embedded database; it was deprecated in version v1.19.1 and is no longer supported. In K3s v1.19.1+, embedded etcd is used.

Reviewer 3: How about we replace the top two paragraphs with:

> As of v1.19.1, embedded etcd is supported. Embedded Dqlite is deprecated.

In the rancher docs, when something is no longer experimental, we just completely remove the word experimental.

Author: Ah, but technically as of v1.19.1 embedded etcd is not supported. It's experimental. Only as of v1.19.5+k3s1 (upcoming release today/tomorrow) is it supported. So should we just say this?

> As of v1.19.5, embedded etcd is supported. Embedded Dqlite is deprecated.

I'm mostly okay with that. We just lose our history here about prior versions technically working with it (just in experimental state).

This is all sort of a mess :/ not sure it makes sense to preserve all that history. Requires a lot of wording around it to try to convey what is what.

Reviewer 3: My reasoning is that from the perspective of someone who is wondering if/when they should upgrade K3s, they probably would only need to know which version it's supported in.

Could ask Craig if we can remove information about when it was experimental.

Author: I'm totally on board with just mentioning when it's supported (I want to call that 1.19.5 not 1.19.1). Everything you said makes sense thanks. I'll see if Craig has time for input. thanks

Reviewer: I like what catherine had originally (plus the word "full"). Its one extra sentence that makes the situation crystal clear. I dont think it detracts at all.

> ### Embedded Dqlite (Deprecated)

Reviewer: Deprecated is not the right word to use. It is full-on removed in 1.19

Can this section be dropped entirely?

Author: +1 on completely dropping the Embedded dqlite section.

> **Warning:** Embedded etcd replaced experimental Dqlite in the K3s v1.19.1 release. Embedded etcd was experimental in the v1.19.1 release and is now considered stable in the 1.19.5 release and newer. This is a breaking change. Please note that upgrades from experimental Dqlite to embedded etcd are not supported. If you attempt an upgrade it will not succeed and data will be lost.

As of v1.0.0, K3s previewed support for running a highly available control plane without the need for an external database.

This architecture is achieved by embedding a Dqlite database within the K3s server process. DQLite is short for "distributed SQLite." According to https://dqlite.io, it is "*a fast, embedded, persistent SQL database with Raft consensus that is perfect for fault-tolerant IoT and Edge devices.*"

> To run K3s with the embedded Dqlite database, follow the same steps as the [embedded etcd database](#embedded-etcd-experimental) using a K3s release between v1.0.0 and v1.19.1.

Author: ```suggestion
To run K3s with the embedded Dqlite database, follow the same steps as the [embedded etcd database](#embedded-etcd) using a K3s release between v1.0.0 and v1.19.1.
```
